src/components/Table/GridTable.tsx

```diff
@@ -260,6 +245,10 @@ export function GridTable<R extends Kinded, X extends Only<GridTableXss, X> = an
     tableState.activeCellId = activeCellId;
   }, [tableState, activeCellId]);
 
+  useEffect(() => {
+    tableState.setSearch(filter);
+  }, [tableState, filter]);
```

Comment:
Pushing the filter into `TableState` so that `RowState` can reactively read it, i.e. parent rows will re-sort their children each time the search changes.